R/parallelization.R

```diff
@@ -1,9 +1,64 @@
 #' Parallelization in purrr
 #'
 #' @description
-#' purrr's map functions have a `.parallel` argument to parallelize a map using
-#' the \CRANpkg{mirai} package. This allows you to run computations in parallel
-#' using more cores on your machine, or distributed over the network.
+#' `r lifecycle::badge("experimental")`
+#'
+#' All map functions allow parallelized operation using \CRANpkg{mirai}. To take
+#' advantage of this, wrap a function that is passed to the `.f` argument of
+#' [map()] or any of its variants with [in_parallel()]. This declares that the
+#' computation should proceed in parallel utilizing multiple cores on your local
+#' machine, or distributed over the network.
+#'
+#' @param .f A fresh formula or function. "Fresh" here means that they should be
+#'   declared in the call to [in_parallel()].
+#' @param ... Arguments to declare in the environment of the function.
+#'
+#' @section How to Use:
+#'
+#' Wrapping a function that is passed to the `.f` argument in [map()] or any of
+#' its variants with [in_parallel()] declares that the map should be performed
+#' in parallel.
+#'
+#' Under the hood, [in_parallel()] provides a systematic way of making `.f`
+#' self-contained so that it can be readily shared with other parallel
+#' processes. It ensures that everything needed by the function is serialized
+#' along with it, but not other objects which happen to be in the function's
+#' enclosing environment. This helps to prevent inadvertently shipping large
+#' data objects when they may not be needed.
+#'
+#' To create self-contained functions:
+#'
+#' * They should call package functions with an explicit `::` namespace. This
+#'   includes packages in the default search path, with the exception of the
+#'   base package. For instance `var()` from the stats package must be called
+#'   with its namespace prefix: `stats::var(x)`.
+#'
+#' * They should declare any data they depend on. You can declare data by
+#'   supplying additional named arguments to `...`⁠.
+#'
+#' [in_parallel()] is a simple wrapper of [carrier::crate()] and you may refer
+#' to that package for more details.
+#'
+#' Example usage:
+#' ```r
```

Comment:
Maybe show the return value of `in_parallel()` and how the size varies as a function of the data that is passed along?

And how the function can be tested on its own to check that every dependency has been packaged up?